src/gui/scheduling/syncscheduler.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#pragma once
+
+#include "etagwatcher.h"
+#include "gui/folder.h"
+
+#include <QObject>
+
+#include <queue>
+#include <unordered_map>
+
+
+namespace OCC {
+
+class FolderMan;
+
+class SyncScheduler : public QObject
+{
+    Q_OBJECT
+public:
+    enum class Priority : uint8_t {
+        // Normal sync triggered by etag change or something similar
+        None,
+
+        // Related to a user action
+        Medium,
+
+        // Usually triggered by a user (ForceSync)
+        Urgent
+    };
+
+    explicit SyncScheduler(FolderMan *parent);
+
+    void enqueueFolder(Folder *folder, Priority priority = Priority::None);
+
+    void start()
+    {
+        _run = true;
+        startNext();
+    }
+
+    void stop() { _run = false; }
+
+    bool isSyncRunning() const { return _currentSync; }
+
+
+private:
+    void startNext();
+
+    bool _run = false;
+    EtagWatcher *_etagManager;
+
+    QPointer<Folder> _currentSync;
+
+    struct QueueInfo
+    {
+        QueueInfo(Folder *f, Priority p);
+
+        // can be deleted
+        QPointer<Folder> folder;
+        // raw pointer for lookup in _scheduledFolders
+        Folder *rawFolder;
```

Comment:
Shouldn't this be equivalent to `folder.data()`? If not, please add a few more words explaining this (apparent, but not necessarily existing) redundancy.